src/ring_theory/integral_closure.lean

```diff
@@ -433,6 +433,132 @@ lemma is_integral.sum {α : Type*} {s : finset α} (f : α → A) (h : ∀ x ∈
   is_integral R (∑ x in s, f x) :=
 (integral_closure R A).sum_mem h
 
+section
+
+variables (p : polynomial R) (x : S)
+
+namespace is_integral_elem_leading_coeff_mul
+
+/-- (Implementation) The monic polynomial whoose root is `f p.leading_coeff * x`. -/
+noncomputable
+def poly (p : polynomial R) : polynomial R :=
```

Comment:
I think `is_integral_elem_leading_coeff_mul.poly` is a reasonable name to suggest that this is merely an auxiliary definition for `is_integral_elem_leading_coeff_mul`?

Reply:
See below for my thoughts on the namespace issue, this may be an auxiliary definition, but why not give it the name of what it does? Over a field this coincides with the composition of the functions `scale_roots` and `normalize` coming from the normalization_monoid structure on polynomials over a field, in that sense there is nothing auxliary about it, you added it for a specific purpose but the definition can have other uses later when your work is generalized or added to, or maybe completely unrelated uses.